hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -269,6 +350,68 @@ public <T> void processNextRecord(HoodieRecord<T> newRecord) throws IOException
     }
   }
 
+  private <T> void processNextNonUniqueKeyRecord(HoodieRecord<T> newRecord) throws IOException {
```

Comment:
I think we should probably subclass /find some way to avoid having this code right inside this class?